internal/pkg/slashing/check.go

```diff
@@ -130,7 +130,7 @@ func verifyBlockSignature(ctx context.Context, view FaultStateView, blk block.Bl
 	if err != nil {
 		panic(errors.Wrapf(err, "failed to inspect miner addresses"))
 	}
-	err = state.NewSignatureValidator(view).ValidateSignature(ctx, blk.SignatureData(), worker, blk.BlockSig)
+	err = state.NewSignatureValidator(view).ValidateSignature(ctx, blk.SignatureData(), worker, *blk.BlockSig)
```

Comment:
`*blk.BlockSig` will panic if nil. Is this guaranteed not to be?

Reply:
I didn't follow up properly on this one, I'll take a look

Reply:
turns out we really need that nil check nice catch